model/repository/MonitoringRepository.php

```diff
@@ -460,23 +460,52 @@ protected function prepareOrderStmt($order): string
     {
         $order = explode(',', $order);
         $result = [];
-        $primaryTableColumns = $this->getPrimaryColumns();
-        foreach ($order as $ruleNum => $orderRule) {
-            preg_match('/([a-zA-Z_][a-zA-Z0-9_]*)\s?(asc|desc)?\s?(string|numeric)?/i', $orderRule, $ruleParts);
+        foreach ($order as $orderRule) {
+            $result[] = $this->buildSingleOrderRule($orderRule);
+        }
+
+        return implode(', ', $result);
+    }
 
-            if (!in_array($ruleParts[1], $primaryTableColumns)) {
-                $colName = $ruleParts[1];
-                $this->queryParams[] = $colName;
+    private function buildSingleOrderRule(string $orderRule): string
+    {
+        if (!preg_match(
+            '/([a-z_][a-z0-9_]*)\s?(asc|desc)?\s?(string|numeric)?/i',
+            $orderRule,
+            $ruleParts
+        )) {
+            return '';
+        }
+
+        $orderBy = $ruleParts[1];
+        $order = $ruleParts[2] ?? 'ASC';
+        $type = $ruleParts[3] ?? null;
+
+        if (!in_array($orderBy, $this->getPrimaryColumns(), true)) {
+            $colName = $orderBy;
+            if (in_array($this->getPlatformName(), ['mysql', 'sqlite'])) {
+                $colName = sprintf('JSON_EXTRACT(t.%s, \'$.%s\')', self::COLUMN_EXTRA_DATA, $colName);
             } else {
-                $sortingColumn = $ruleParts[1];
+                $colName = sprintf('t.%s ->> \'%s\'', self::COLUMN_EXTRA_DATA, $colName);
             }
-
-            $result[] = isset($ruleParts[3]) && $ruleParts[3] === 'numeric'
-                ? sprintf("cast(nullif(%s, '') as decimal) %s", $sortingColumn, $ruleParts[2])
-                : sprintf('%s %s', $sortingColumn, isset($ruleParts[2]) ? $ruleParts[2] : 'ASC');
+            $sortingColumn = $colName;
+        } else {
+            $sortingColumn = $orderBy;
         }
 
-        return implode(', ', $result);
+        return isset($type) && $type === 'numeric'
```

Comment:
```suggestion
        return $type === 'numeric'
```

This would be sufficient, no need to have two conditions here.